axelor-sale/src/main/java/com/axelor/apps/sale/service/saleorder/SaleOrderLineServiceImpl.java

```diff
@@ -262,7 +270,8 @@ protected BigDecimal fillDiscount(
             taxService.convertUnitPrice(
                 saleOrderLine.getProduct().getInAti(),
                 saleOrderLine.getTaxLine(),
-                (BigDecimal) discounts.get("discountAmount")));
+                (BigDecimal) discounts.get("discountAmount"),
+                AppBaseService.COMPUTATION_SCALING));
```

Comment:
it looks like it is directly set in the `discountAmount` field in sale order line here, so please use the scale for unit price.